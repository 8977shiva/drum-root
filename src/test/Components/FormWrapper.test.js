import React from "react";
import { shallow } from "enzyme";
import FormWrapper from "../../Components/FormWrapper";

describe("FormWrapper component", () => {
  let wrapper;
  const onSubmit = jest.fn(() => Promise.resolve());

  beforeEach(() => {
    wrapper = shallow(<FormWrapper onSubmit={onSubmit} />);
  });
  test("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("onSubmit prop should be called when submitting form", () => {
    wrapper
      .find("form")
      .at(0)
      .prop("onSubmit")();
    expect(onSubmit).toHaveBeenCalled();
  });
});
