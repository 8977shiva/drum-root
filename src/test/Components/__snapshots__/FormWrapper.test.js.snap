// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormWrapper component should render properly 1`] = `
<form
  onSubmit={[Function]}
/>
`;
